app/init/websocket_manager.ts

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import NetInfo, {NetInfoState} from '@react-native-community/netinfo';
+import {AppState, AppStateStatus} from 'react-native';
+
+import {setCurrentUserStatusOffline} from '@actions/local/user';
+import {fetchStatusByIds} from '@actions/remote/user';
+import {handleClose, handleEvent, handleFirstConnect, handleReconnect} from '@actions/websocket';
+import WebSocketClient from '@app/client/websocket';
+import {General} from '@app/constants';
+import {queryWebSocketLastDisconnected} from '@app/queries/servers/system';
+import {queryAllUsers} from '@app/queries/servers/user';
+import DatabaseManager from '@database/manager';
+
+import type {ServerCredential} from '@typings/credentials';
+
+class WebsocketManager {
+    private clients: Record<string, WebSocketClient> = {};
+    private statusUpdatesIntervalIDs: Record<string, NodeJS.Timer> = {};
+    private previousAppState: AppStateStatus;
+    private netConnected = false;
+
+    constructor() {
+        this.previousAppState = AppState.currentState;
+    }
+
+    public init = async (serverCredentials: ServerCredential[]) => {
+        this.netConnected = Boolean((await NetInfo.fetch()).isConnected);
+        await Promise.all(
+            serverCredentials.map(
+                async ({serverUrl, token}) => {
+                    const database = DatabaseManager.serverDatabases[serverUrl]?.database;
+                    if (!database) {
+                        return;
```

Comment:
shouldn't we continue so we can connect the other valid ones ?

Reply:
We are returning from the function called by map, so we are only returning for this pair `{serverUrl, token}`. The rest of the of the pairs of `{serverUrl, token}` will keep executing thanks to the map.